checks.d/yarn.py

```diff
@@ -146,61 +150,60 @@ def check(self, instance):
-        self._yarn_app_metrics(rm_address)
-        self._yarn_node_metrics(cluster_id, rm_address)
+        # Get additional tags from the conf file
+        tags = instance.get('tags', [])
+        if tags is None:
+            tags = []
+        else:
+            tags = list(set(tags))
 
-    def _get_cluster_id(self, rm_address):
-        '''
-        Return the cluster ID, otherwise raise an exception
-        '''
-        info_json = self._rest_request_to_json(rm_address, INFO_URI)
+        # Get the cluster name from the conf file
+        cluster_name = instance.get('cluster_name')
+        if cluster_name is None:
+            raise Exception('The cluster_name must be specified in the instance configuration')
 
-        cluster_id = info_json.get('clusterInfo', {}).get('id')
-        if cluster_id is not None:
-            return cluster_id
+        tags.append('cluster_name:%s' % cluster_name)
 
-        raise Exception('Unable to retrieve cluster ID from ResourceManger')
+        # Get metrics from the Resource Manager
+        self._yarn_cluster_metrics(rm_address, tags)
+        self._yarn_app_metrics(rm_address, tags)
+        self._yarn_node_metrics(rm_address, tags)
 
-    def _yarn_cluster_metrics(self, cluster_id, rm_address):
+    def _yarn_cluster_metrics(self, rm_address, addl_tags):
         '''
         Get metrics related to YARN cluster
         '''
         metrics_json = self._rest_request_to_json(rm_address, YARN_CLUSTER_METRICS_PATH)
 
         if metrics_json:
-            tags = ['cluster_id:' + str(cluster_id)]
 
             yarn_metrics = metrics_json[YARN_CLUSTER_METRICS_ELEMENT]
 
             if yarn_metrics is not None:
-                self._set_yarn_metrics_from_json(tags, yarn_metrics, YARN_CLUSTER_METRICS)
+                self._set_yarn_metrics_from_json(addl_tags, yarn_metrics, YARN_CLUSTER_METRICS)
 
-    def _yarn_app_metrics(self, rm_address):
+    def _yarn_app_metrics(self, rm_address, addl_tags):
         '''
-        Get metrics related to YARN applications
+        Get metrics for running applications
         '''
-        metrics_json = self._rest_request_to_json(rm_address, YARN_APPS_PATH)
+        metrics_json = self._rest_request_to_json(rm_address,
+            YARN_APPS_PATH,
+            states=YARN_APPLICATION_STATES)
 
         if metrics_json:
             if metrics_json['apps'] is not None:
                 if metrics_json['apps']['app'] is not None:
 
                     for app_json in metrics_json['apps']['app']:
 
-                        cluster_id = app_json['clusterId']
-                        app_id = app_json['id']
+                        app_name = app_json['name']
 
-                        tags = ['cluster_id:' + str(cluster_id), 'app_id:' + str(app_id)]
+                        tags = ['app_name:%s' % str(app_name)]
+                        tags.extend(addl_tags)
 
                         self._set_yarn_metrics_from_json(tags, app_json, YARN_APP_METRICS)
```

Comment:
The check now tags these metrics by `app_name` instead of `app_id`, so there can be duplicate sets of tags for the same metric (when multiple apps have the same name). So we should use `increment` instead of `gauge` to send these metrics